apps/examples/testcase/le_tc/network/itc_net_connect.c

```diff
@@ -263,26 +264,26 @@ static void itc_net_connect_p(void)
 	int num_clients = 1;
 	int id = 1;
 	int ret;
-	int *ptr = &num_clients;
-	int *ptr_id = &id;
 	int *pret = OK;
-	g_port = 8880;
 
-	pthread_create(&server_thread, NULL, server_connect, ptr);
-	sleep(SERVER_WAIT_TIME);
-	pthread_create(&client_thread, NULL, client_connect, ptr_id);
+	g_port = 8880;
+	_initialize_test();
 
-	sleep(CLIENT_WAIT_TIME);
+	pthread_create(&server_thread, NULL, server_connect, (void *)&num_clients);
+	sem_wait(&g_sig_srvtomain);
+	pthread_create(&client_thread, NULL, client_connect, (void *)&id);
 
 	ret = pthread_join(server_thread, (void *)&pret);
 	TC_ASSERT_EQ_CLEANUP("pthread_join", ret, OK, free(pret); pthread_join(client_thread, NULL));
 	TC_ASSERT_EQ_CLEANUP("pthread_join", *pret, OK, free(pret); pthread_join(client_thread, NULL));
```

Comment:
Is it okay to use '*pret' even pthread returns NULL value?

Reply:
You're right. it's possible to free NULL if server_thread return NULL. It seems difficult problem. So I'll modify it next PR. Thanks for your analysis.

Reply:
Okay, I see